core/src/test/java/org/hibernate/ogm/backendtck/innertypes/InnerClassFindTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.innertypes;
+
+import static org.fest.assertions.Assertions.assertThat;
+import javax.persistence.EntityManager;
+import org.hibernate.ogm.backendtck.innertypes.CommunityMember.Employee;
+import org.hibernate.ogm.utils.TestForIssue;
+import org.hibernate.ogm.utils.jpa.JpaTestCase;
+import org.junit.Test;
+
+/**
+ * Verifies the dialect is able to store an entity which is represented as an inner type in Java.
+ *
+ * For example on MongoDB this implies the name of the class needs to avoid the dollar symbol,
+ * so this needs to be adjusted by forcing a valid name using the JPA Table annotation.
+ *
+ * See also org.hibernate.ogm.datastore.mongodb.test.datastore.CollectionNamingValidationTest in the MongoDB module.
+ *
+ * @author Davide D'Alto
+ * @author Sanne Grinovero
+ */
+@TestForIssue(jiraKey = "OGM-265")
+public class InnerClassFindTest extends JpaTestCase {
+
+	@Test
+	public void testInnerClassFind() throws Exception {
```

Comment:
> That's not about testing this contract is it?

It's about making sure that the SPI does what it promises. This includes having tests which show that the SPI is invoked at the right places. At least the most-standard scenarios should be covered, to avoid a huge WT* moment for users wishing to provide an SPI implementation and finding out it it's not invoked in a common case.